langkit/themes.py

```diff
@@ -15,44 +15,52 @@
 
 _transformer_model = None
 _theme_groups = None
-
 lang_config = LangKitConfig()
 
+_jailbreak_embeddings = None
+_refusal_embeddings = None
+
 
 def register_theme_udfs():
+    global _jailbreak_embeddings
+    global _refusal_embeddings
+
+    _jailbreak_embeddings = [
+        _transformer_model.encode(s, convert_to_tensor=True)
+        for s in _theme_groups.get("jailbreaks", [])
+    ]
+    _refusal_embeddings = [
+        _transformer_model.encode(s, convert_to_tensor=True)
+        for s in _theme_groups.get("refusals", [])
+    ]
+
     if "jailbreaks" in _theme_groups:
-        jailbreak_embeddings = [
-            _transformer_model.encode(s, convert_to_tensor=True)
-            for s in _theme_groups["jailbreaks"]
-        ]
-
-        @register_metric_udf(col_type=String)
-        def jailbreak_similarity(text: str) -> float:
-            if _transformer_model is None:
-                raise ValueError("Must initialize a transformer before calling encode!")
-            similarities = []
-            text_embedding = _transformer_model.encode(text, convert_to_tensor=True)
-            for embedding in jailbreak_embeddings:
-                similarity = get_embeddings_similarity(text_embedding, embedding)
-                similarities.append(similarity)
-            return max(similarities)
+        register_metric_udf(col_type=String)(jailbreak_similarity)
 
     if "refusals" in _theme_groups:
-        refusal_embeddings = [
-            _transformer_model.encode(s, convert_to_tensor=True)
-            for s in _theme_groups["refusals"]
-        ]
-
-        @register_metric_udf(col_type=String)
```

Comment:
We might want to log an info/warning when the refusals or jailbreaks are not in the _theme_groups, it seems like someone might accidentally miss adding one of these and in that case its a kind of silent failure.